client/state/test/utils.js

```diff
@@ -233,17 +229,15 @@ describe( 'utils', () => {
 			} );
 
 			test( 'should return current state when serialize action type passed', () => {
-				expect( reducer( currentState, actionSerialize ) ).to.be.deep.equal( currentState );
+				expect( reducer( currentState, actionSerialize ) ).toBe( currentState );
```

Comment:
`to.be.deep.equal` translates to `toEqual` -- but I guess you're picking `toBe` since we're really about unmodified state, i.e. referential equality?

Reply:
😮 That was a surprise for me, nice spot. So `.deep.equal` === `.eql` === `.eqls`.

> ### `.deep`
> Causes all `.equal`, `.include`, `.members`, `.keys`, and `.property` assertions that follow in the chain to use deep equality instead of strict (`===`) equality. See the `deep-eql` project page for info on the deep equality algorithm: https://github.com/chaijs/deep-eql.

[#](http://www.chaijs.com/api/bdd/#method_deep)

> The `.deep.equal` assertion is almost identical to `.eql` but with one difference: `.deep.equal` causes deep equality comparisons to also be used for any other assertions that follow in the chain.

That's a bit confusing 🙄 

I had been translateing `eql` -> `toEqual` and `equal` -> `toBe`, because (without `deep`) they correspond to `==` and `===`, respectively.

---

Now that I'm actually considering the implications of this change, of which I was unaware, `toBe` seems like the right choice. That's because `currentState` is deeply-frozen. If the objects are referentially equal, I'd expect them to be deeply equal as well:

https://github.com/Automattic/wp-calypso/blob/78fd14f8c2ea00a17dcb0f65ccf095d4966228a9/client/state/test/utils.js#L29-L31

Reply:
> Now that I'm actually considering the implications of this change, of which I was unaware, `toBe` seems like the right choice. That's because `currentState` is deeply-frozen. If the objects are referentially equal, I'd expect them to be deeply equal as well:

That's fair. Personally, I'd probably still prefer `.toEqual` here, since state being deep frozen could be considered a test implementation detail that could get inadvertently lost. Your call tho :smile: